samples/client/petstore/rust-reqwest/src/apis/pet_api.rs

```diff
@@ -0,0 +1,349 @@
+/* 
+ * OpenAPI Petstore
+ *
+ * This is a sample server Petstore server. For this sample, you can use the api key `special-key` to test the authorization filters.
+ *
+ * OpenAPI spec version: 1.0.0
+ * 
+ * Generated by: https://openapi-generator.tech
+ */
+
+use std::rc::Rc;
+use std::borrow::Borrow;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use serde_json;
+use serde_json::Value;
+
+use reqwest;
+
+use super::{Error, configuration};
+
+pub struct PetApiClient {
+    configuration: Rc<configuration::Configuration>,
+}
+
+impl PetApiClient {
+    pub fn new(configuration: Rc<configuration::Configuration>) -> PetApiClient {
+        PetApiClient {
+            configuration: configuration,
+        }
+    }
+}
+
+pub trait PetApi {
+    fn add_pet(&self, pet: ::models::Pet) -> Result<(), Error>;
+    fn delete_pet(&self, pet_id: i64, api_key: &str) -> Result<(), Error>;
+    fn find_pets_by_status(&self, status: Vec<String>) -> Result<Vec<::models::Pet>, Error>;
+    fn find_pets_by_tags(&self, tags: Vec<String>) -> Result<Vec<::models::Pet>, Error>;
+    fn get_pet_by_id(&self, pet_id: i64) -> Result<::models::Pet, Error>;
+    fn update_pet(&self, pet: ::models::Pet) -> Result<(), Error>;
+    fn update_pet_with_form(&self, pet_id: i64, name: &str, status: &str) -> Result<(), Error>;
+    fn upload_file(&self, pet_id: i64, additional_metadata: &str, file: ::models::File) -> Result<::models::ApiResponse, Error>;
+}
+
+
+impl PetApi for PetApiClient {
+    fn add_pet(&self, pet: ::models::Pet) -> Result<(), Error> {
+        let configuration: &configuration::Configuration = self.configuration.borrow();
+        let client = &configuration.client;
+
+
+
+        let query_string = {
+            let mut query = ::url::form_urlencoded::Serializer::new(String::new());
+
+            query.finish()
+        };
+        let uri_str = format!("{}/pet?{}", configuration.base_path, query_string);
```

Comment:
It seems as if when there are no query parameters we construct and provide an empty query string. I'm not sure it's wrong, but it seems a little unusual.

Reply:
Yes. I noticed it too. It comes from the forked `hyper` code. I did not take time to handle this better for now. It is something I correct manualy post-generation. But there is probably something to do.